src/components/UploadPage/components/UploadErrorReport/UploadErrorReport.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { ErrorReport, GetErrorReportKeys } from "../../../../models/FileReport";
+import { ErrorLabels } from "../../../../util/Constants";
+
+interface Props {
+  errorReport: ErrorReport;
+  filename: string;
+  reportIndex: number;
+}
+
+const UploadErrorReport: React.FC<Props> = ({
+  errorReport,
+  filename,
+  reportIndex
+}) => {
+  const [isOpen, setOpen] = React.useState(true);
+
+  const onToggle = () => {
+    setOpen(!isOpen);
+  };
+
+  return (
+    <div className="upload-error-report">
+      <div
+        className={`upload-error-report__header ${
+          isOpen ? "" : "upload-error-report__header--closed"
+        }`}
+        onClick={onToggle}
+      >
+        Error
+        <i className={`upload-error-report__arrow ${isOpen ? "down" : "up"}`} />
+      </div>
+      {isOpen ? (
+        <div className="upload-error-report__body">
+          {GetErrorReportKeys(errorReport)
+            .filter((errorKey: string) => errorKey !== "error_type")
+            .map((errorKey, errorIndex: number) => (
+              <div
+                key={`upload-error-report-key-${filename}-${reportIndex}-${errorIndex}`}
+                className="upload-error-report__error-row"
+              >
+                <div className="upload-error-report__error-label">
+                  {ErrorLabels[errorKey]}:
+                </div>
+                <div className="upload-error-report__error-data">
+                  {errorReport[errorKey].toString()}
```

Comment:
So there's an issue occurring here if I try to check if the `errorReport[errorKey]` is of type `Array<number>`. `ErrorReport` is a type combined of all seven error possibilities and really doesn't seem to be aware of the possibility that the value here could be `list_of_positions` when I try `Array.isArray()`. The use of `toString()` is a stop gap measure but do you know a better way to make the function aware of the value?

Reply:
Further explanation is that it seems that in case of combined types, the TypeScript only looks at the lowest common denominators. In this case, `error_type` and `message` are the only keys that all error types possess:

![image](https://user-images.githubusercontent.com/4026454/72303218-7eb4c280-363a-11ea-816d-61d2bee96b40.png)

Reply:
Tried Type Guard with `(errorReport as UnmatchedParens)` but it doesn't seem to jive either...

Reply:
Splitting individual components per type variety works without complaint but it's pretty verbose. Can show you tomorrow.